packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -449,6 +453,20 @@ Future<void> diagnoseXcodeBuildFailure(XcodeBuildResult result) async {
     ).send();
   }
 
+  // Building for iOS Simulator, but the linked and embedded framework 'App.framework' was built for iOS.
+  // or
+  // Building for iOS, but the linked and embedded framework 'App.framework' was built for iOS Simulator.
+  if (result.stdout?.contains('Building for iOS') == true
+      && result.stdout?.contains('but the linked and embedded framework') == true
+      && result.stdout?.contains('was built for iOS') == true) {
+    globals.printError('');
+    globals.printError('Your Xcode project requires migration. See https://github.com/flutter/flutter/issues/50568 for details.');
```

Comment:
This link is temporary.  I will write wiki or website instructions and update this link.